src/main/java/erb/RouteImpl.java

```diff
@@ -59,5 +74,52 @@ public void setDuration(String d) {
 		this.duration = d;
 	}
 
+	// erb: vk US-55-TASK-11 - follow mc and set the sort tag for sorting columns of the route
+	@Override
+	public void setSortTag(int sortTag) {
+		this.sortTag = sortTag;
+
+	}
+
+	// erb: vk US-55-TASK-11 - follow mc and getting the sort tag for sorting columns of the route
+	public int getSortTag() {
+		return sortTag;
+	}
 	
+	/**
+	 *  Method: compareTo
+	 *  Inputs: route object
+	 *  Returns: @param compareResult 
+	 *  
+	 *  Description:  // erb: vk US-55-TASK-112 - adding override for compareTo from class Comparable, 
+	 *  compares object in order to sort columns properly in the UI
+	 */  
+	@Override
+	public int compareTo(Route other) {
+			if(sortTag == Driver.SORTBY_ID) {
+				//sort using long 
+				return (Long.valueOf(routeId)).compareTo((Long.valueOf(other.getRouteId())));
+			}
+			else if(sortTag == _SORTBY_ROUTE_NAME) {
+				return routeName.compareTo(other.getRouteName());
+			}
+			else if(sortTag == _SORTBY_LENGTH) {
+				//check if same first name, sort by last name if so.
+				int compareVal = length.compareTo(other.getLength());
+				if(compareVal == 0) {
+					return duration.compareTo(other.getDuration());
+				}
+				return compareVal;
+			}
+			else if(sortTag == _SORTBY_DURATION) {
+				int compareVal = duration.compareTo(other.getDuration());
```

Comment:
Need to convert to a numerical value to sort properly and could remove the extra comparisons.  
```suggestion
				return  (Double.valueOf(duration).compareTo(Double.valueOf(other.getDuration()));
```

Reply:
All done:  changed as follows:
	             return (Double.valueOf(length)).compareTo((Double.valueOf(other.getLength())));